pkg/api/v1/load_balancers_create.go

```diff
@@ -32,63 +34,69 @@ func (r *Router) loadBalancerCreate(c echo.Context) error {
 		return v1BadRequestResponse(c, err)
 	}
 
-	lbs := models.LoadBalancerSlice{}
-
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
 		r.logger.Errorw("failed to begin transaction", "error", err)
 		return v1BadRequestResponse(c, err)
 	}
 
-	for _, p := range payload {
-		lb := &models.LoadBalancer{
-			TenantID:         tenantID,
-			DisplayName:      p.DisplayName,
-			LoadBalancerSize: p.LoadBalancerSize,
-			LoadBalancerType: p.LoadBalancerType,
-			IPAddr:           p.IPAddr,
-			LocationID:       p.LocationID,
-			Slug:             slug.Make(p.DisplayName),
-			CurrentState:     "provisioning",
-		}
-
-		if err := validateLoadBalancer(lb); err != nil {
-			_ = tx.Rollback()
-
-			r.logger.Errorw("failed to validate load balancer", "error", err)
+	lb := &models.LoadBalancer{
+		TenantID:         tenantID,
+		DisplayName:      payload.DisplayName,
+		LoadBalancerSize: payload.LoadBalancerSize,
+		LoadBalancerType: payload.LoadBalancerType,
+		IPAddr:           payload.IPAddr,
+		LocationID:       payload.LocationID,
+		Slug:             slug.Make(payload.DisplayName),
+		CurrentState:     "provisioning",
+	}
 
-			return v1UnprocessableEntityResponse(c, err)
-		}
+	if err := validateLoadBalancer(lb); err != nil {
+		_ = tx.Rollback()
 
-		lbs = append(lbs, lb)
+		r.logger.Errorw("failed to validate load balancer", "error", err)
 
-		err = lb.Insert(ctx, tx, boil.Infer())
-		if err != nil {
-			r.logger.Errorw("failed to create load balancer, rolling back transaction", "error", err)
+		return v1UnprocessableEntityResponse(c, err)
+	}
 
-			if err := tx.Rollback(); err != nil {
-				r.logger.Errorw("failed to rollback transaction", "error", err)
-				return v1InternalServerErrorResponse(c, err)
-			}
+	err = lb.Insert(ctx, tx, boil.Infer())
+	if err != nil {
+		r.logger.Errorw("failed to create load balancer, rolling back transaction", "error", err)
 
+		if err := tx.Rollback(); err != nil {
+			r.logger.Errorw("failed to rollback transaction", "error", err)
 			return v1InternalServerErrorResponse(c, err)
 		}
+
+		return v1InternalServerErrorResponse(c, err)
 	}
 
-	switch len(lbs) {
-	case 0:
-		if err := tx.Rollback(); err != nil {
-			r.logger.Errorw("failed to rollback transaction", "error", err)
-			return v1BadRequestResponse(c, err)
-		}
+	if err := tx.Commit(); err != nil {
```

Comment:
Same question about the transaction in this case.  Not sure if it makes the select below deterministic or of its unnecessary?